src/Core/ServiceWrapper/Main.cs

```diff
@@ -542,8 +557,8 @@ public static void Run(string[] _args)
                     args = args.GetRange(2, args.Count - 2);
                 }
 
-                args[0] = args[0].ToLower();
-                if (args[0] == "install")
+                String command = args[0].ToLower();
+                if (command == "install")

```

Comment:
The solution occasionally adds the refactoring of the commands handling.
It needs to be reworked to have a proper merge with https://github.com/kohsuke/winsw